### ISSUE TITLE
[3.9.14] Article filter does not display all categories, even for a super admin

### ISSUE THREAD

**Issue by Gnevyshev** [joomla/joomla-cms#27509]
### Steps to reproduce the issue
Add new access level Test
Add content category TestCat with access level Test
Add article TestArticle with category TestCat and eccess level Test
Go to Content - Articles
Click Search Tools, cursor to Select Category

### Expected result
All categories in Select Category (I'm super admin)

### Actual result
Only categories with access level Public

### System information (as much as possible)
Joomla! 3.9.14
PHP 7.2

### Additional comments
For category TestCat permissions for super users - all allowed

**Comment by ReLater** [comment]
You must select at least one user group for the access level. See checkboxes "User Groups Having Viewing Access".

**Comment by Gnevyshev** [comment]
But why?
We can create an access level without a group. And this is right. And I use it.
After all, categories with this level of access are displayed in the Category list.
Why are they not displayed in this filter?

**Comment by joomdonation** [comment]
Yes, it was fixed with this PR https://github.com/joomla/joomla-cms/pull/40193 . Thanks for checking @krishnagandhicode
